sqlglot/dialects/risingwave.py

```diff
@@ -1,6 +1,501 @@
+            unique = " UNIQUE" if expression.args.get("unique") else ""
+
+            clustered = expression.args.get("clustered")
+            if clustered is None:
+                clustered_sql = ""
+            elif clustered:
+                clustered_sql = " CLUSTERED COLUMNSTORE"
+            else:
+                clustered_sql = " NONCLUSTERED COLUMNSTORE"
+
+            postcreate_props_sql = ""
+            if properties_locs.get(exp.Properties.Location.POST_CREATE):
+                postcreate_props_sql = self.properties(
+                    exp.Properties(
+                        expressions=properties_locs[exp.Properties.Location.POST_CREATE]
+                    ),
+                    sep=" ",
+                    prefix=" ",
+                    wrapped=False,
+                )
+
+            modifiers = "".join((clustered_sql, replace, refresh, unique, postcreate_props_sql))
+
+            postexpression_props_sql = ""
+            if properties_locs.get(exp.Properties.Location.POST_EXPRESSION):
+                postexpression_props_sql = self.properties(
+                    exp.Properties(
+                        expressions=properties_locs[exp.Properties.Location.POST_EXPRESSION]
+                    ),
+                    sep=" ",
+                    prefix=" ",
+                    wrapped=False,
+                )
+
+            if kind == "SINK" or kind == "SOURCE" and self.pretty:
+                postexpression_props_sql = self.sep() + postexpression_props_sql.strip()
+
+            concurrently = " CONCURRENTLY" if expression.args.get("concurrently") else ""
+            exists_sql = " IF NOT EXISTS" if expression.args.get("exists") else ""
+            no_schema_binding = (
+                " WITH NO SCHEMA BINDING" if expression.args.get("no_schema_binding") else ""
+            )
+
+            clone = self.sql(expression, "clone")
+            clone = f" {clone}" if clone else ""
+
+            if kind == "SINK":
+                # Tailored for risingwave sink.
+                expression_sql = f"CREATE SINK{concurrently}{exists_sql} {this}{expression_sql}{properties_sql}{postexpression_props_sql}{index_sql}{no_schema_binding}{clone}"
+            else:
+                expression_sql = f"CREATE{modifiers} {kind}{concurrently}{exists_sql} {this}{properties_sql}{expression_sql}{postexpression_props_sql}{index_sql}{no_schema_binding}{clone}"
+            return self.prepend_ctes(expression, expression_sql)
+
+        def sql(
+            self,
+            expression: t.Optional[str | exp.Expression],
+            key: t.Optional[str] = None,
+            comment: bool = True,
+        ) -> str:
+            if isinstance(expression, exp.Watermark):
+                sql = self._watermark_sql(expression)
+                return self.maybe_comment(sql, expression) if self.comments and comment else sql
+
+            return super().sql(expression, key, comment)
```

Comment:
Delete this and rename `_watermark_sql` to `watermark_sql`.

Reply:
Do you mean deleting this `sql` override?

Reply:
Yep.

Reply:
Then how to parse watermark related sql?

Reply:
What does parsing have to do with this override? This is in the generator class. The suggestion has to do with how the watermark expression is generated: you shouldn't have to override `sql` in order to specify how an expression is converted in to SQL, you won't see this pattern anywhere in the codebase. Instead, you do `<expression_name_all_lowercase>_sql` and that is automatically used to generate the corresponding expression, hence my suggestion in the first message in this thread.